rotkehlchen/chain/ethereum/modules/uniswap/v3/utils.py

```diff
@@ -0,0 +1,557 @@
+                    f'due to: {str(e)}',
+                )
+                continue
+        # Get the ranges of price for which each position is valid for.
+        # Get the amount of each token present in an LP position.
+        price_ranges = []
+        amounts_0 = []
+        amounts_1 = []
+        for (position, slot_0, token_a, token_b) in zip(positions, slots_0, tokens_a, tokens_b):
+            price_ranges.append(
+                calculate_price_range(
+                    tick_lower=position[5],
+                    tick_upper=position[6],
+                    decimal_0=token_a['decimals'],
+                    decimal_1=token_b['decimals'],
+                ),
+            )
+            amounts_0.append(
+                calculate_amount(
+                    tick_lower=position[5],
+                    liquidity=position[7],
+                    tick_upper=position[6],
+                    decimals=token_a['decimals'],
+                    tick=slot_0[1],
+                    token_position=0,
+                ),
+            )
+            amounts_1.append(
+                calculate_amount(
+                    tick_lower=position[5],
+                    liquidity=position[7],
+                    tick_upper=position[6],
+                    decimals=token_b['decimals'],
+                    tick=slot_0[1],
+                    token_position=1,
+                ),
+            )
+        # First, get the total liquidity of the LPs.
+        # Use the value of the liquidity to get the total amount of tokens in LPs.
+        total_tokens_in_pools = []
+        try:
+            liquidity_in_pools_multicall = multicall_2(
+                ethereum=ethereum,
+                require_success=False,
+                calls=[
+                    (entry[0], entry[1].encode('liquidity'))
+                    for entry in zip(pool_addresses, pool_contracts)
+                ],
+            )
+        except RemoteError as e:
+            log.error(UNISWAP_V3_ERROR_MSG.format('pool contract liquidity', str(e)))
+            continue
+
+        for _entry in zip(
+            pool_contracts,
+            liquidity_in_pools_multicall,
+            positions,
+            slots_0,
+            tokens_a,
+            tokens_b,
+        ):
+            liquidity_in_pool = _entry[0].decode(_entry[1][1], 'liquidity')[0]
+            try:
+                total_tokens_in_pools.append(
+                    calculate_total_amounts_of_tokens(
+                        liquidity=liquidity_in_pool,
+                        tick=_entry[3][1],
+                        fee=_entry[2][4],
+                        decimal_0=_entry[4]['decimals'],
+                        decimal_1=_entry[5]['decimals'],
+                    ),
+                )
+            except UnrecognizedFeeTierException as e:
+                log.error(f'Error calculating total amount of tokens in pool due to: {str(e)}')
```

Comment:
But this won't be seen by the user. Send it to the user via the msg_aggregator.

Reply:
Okay